headerfs/file_test.go

```diff
@@ -0,0 +1,179 @@
+package headerfs
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestAppendRow(t *testing.T) {
+	tests := []struct {
+		name          string
+		initialData   []byte
+		headerToWrite []byte
+		writeFn       func([]byte, FileInterface) (int, error)
+		truncFn       func(size int64) error
+		expected      []byte
+		wantErr       bool
+		errMsg        string
+	}{
+		{
+			name:          "NormalWrite_ValidHeader_DataAppendedSuccessfully",
```

Comment:
Fwiw if you add spaces in the naem here, then the `go test` tool will automatically add underscores there.